### DISCUSSION TITLE
How about showing full article content in RSS feed?

### DISCUSSION THREAD

**Discussion by schulle4u** [datenstrom/community#228] [Ask a question] [answered]
I'd like to add an option to the feed plugin which allows to display the full content of a page or an article in the RSS output. The resulting item's section in the feed should look something like this: 

```
<description><![CDATA[Description or excerpt goes here]]></description>
<content><![CDATA[Full article content for supported readers]]></content>
```

Can this already be done using the yellow API? Just using the `$yellow->page->getContent()` function probably is not the right way to parse content into a feed as there are elements that should be stripped out I think. 

Steffen

**Comment by markseuffert** [comment]
Hi Steffen, I think this is possible by changing the `FeedPageLength` to a very large number.
For example add `FeedPageLength: 1234000` to your `system/config/config.ini` file.

Hope this helps.

**Comment by schulle4u** [comment]
Hi mark, 

yes, that's of course the quick and dirty way. :) I've used this method already. My idea was to make the feed a little more standard-compliant by providing separate description and content sections in the feed.

Steffen

**Comment by markseuffert** [comment]
That's a good idea. It makes sense to offer both elements in the RSS feed. I think you can either keep HTML elements or remove them from the content, as described in the [RSS specification](http://www.rssboard.org/rss-profile#namespace-elements-content). In the Yellow API there's the function `createTextDescription()`, which you can use to remove HTML elements.

Let me know if you improve the feed plugin. 😄

**Comment by schulle4u** [comment]
I've made a working attempt. 

https://github.com/schulle4u/yellow-plugins/blob/schulle4u-patch-1/feed/feed.php

Use `feedType: full`in config.ini to see this in action. I'm using the `content:encoded` tag, which specifically is designed to display HTML content in supported feed readers. 

Feel free to make further suggestions. And if my modifications could be useful, I'll prepare a pull request. 

Steffen

**Comment by markseuffert** [comment, marked as answer]
I merged your code and updated blog and feed plugin.  Now they both show the full page content, unless there's a manual page break. You can add `[--more--]` where your want the break. Probably not a surprise for you and me, just wanted to make clear how users can control their blog.

Let me know if this works for you as expected.

**Comment by schulle4u** [comment]
Sorry for being late. 
Both blog and feed plugins work as expected, I've just updated to the latest version on one of my sites. Thanks for officially implementing this feature. :)

Steffen
